### PR TITLE
(SLV-365) Update the install, configure, and upgrade plans in the pe_xl module to make ha optional

### PR DESCRIPTION
**Pull request by billclaytor**
This update adds an optional `ha` parameter (default = true) and conditionally performs the HA steps in the install, configure, and upgrade plans.

Once this update has been reviewed internally by the SLV team I will create a new PR for submission to the reidmv/reidmv-pe_xl repo.

~~TODO: Update the [install_and_configure_without_ha.md](https://github.com/billclaytor/reidmv-pe_xl/blob/SLV-365/documentation/install_and_configure_without_ha.md) document on the SLV-365 branch and include it in this branch.~~

### CODE REVIEW

File: plans/configure.pp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # @summary Configure first-time classification and HA setup
 #
 plan pe_xl::configure (
+  Boolean             $ha,
```
Comment (johnduarte) [inline plans/configure.pp]: Should this default to true?

File: plans/upgrade.pp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # @summary Upgrade an Extra Large stack from one .z to the next
 #
 plan pe_xl::upgrade (
+  Boolean   $ha,
```
Comment (johnduarte) [inline plans/upgrade.pp]: Should this default to true?

**Review by johnduarte — APPROVED** [review]

File: plans/configure.pp (hunk 1 above)
Comment (billclaytor) [reply plans/configure.pp]: Updated

File: plans/upgrade.pp (hunk 2 above)
Comment (billclaytor) [reply plans/upgrade.pp]: Updated

File: plans/install.pp [hunk 3]
```diff
@@ -184,38 +244,51 @@
...
 
   # Deploy the PE agent to all remaining hosts
-  run_task('pe_xl::agent_install', $master_replica_host,
-    server        => $master_host,
-    install_flags => [
+  if $ha {
+    run_task('pe_xl::agent_install', $master_replica_host,
+      server        => $master_host,
+      install_flags => [
       '--puppet-service-ensure', 'stopped',
```
Comment (RandellP) [inline plans/install.pp]: these lines need an indent 251-255

File: plans/install.pp [hunk 4]
```diff
@@ -115,16 +155,18 @@
...
   )
 
-  run_task('pe_xl::mkdir_p_file', $puppetdb_database_replica_host,
-    path    => '/etc/puppetlabs/puppet/csr_attributes.yaml',
-    content => @("HEREDOC"),
+  if $ha {
+    run_task('pe_xl::mkdir_p_file', $puppetdb_database_replica_host,
+      path    => '/etc/puppetlabs/puppet/csr_attributes.yaml',
+      content => @("HEREDOC"),
       ---
```
Comment (RandellP) [inline plans/install.pp]: I thin lines 162-167 should be indented one more level.

File: plans/install.pp [hunk 5]
```diff
@@ -184,38 +244,51 @@
...
+      ],
+    )
 
-  run_task('pe_xl::agent_install', $cm_cluster_b,
-    server        => $master_host,
-    install_flags => [
+    run_task('pe_xl::agent_install', $cm_cluster_b,
+      server        => $master_host,
+      install_flags => [
       '--puppet-service-ensure', 'stopped',
```
Comment (RandellP) [inline plans/install.pp]: 273-274 add indent

File: plans/install.pp [hunk 6]
```diff
@@ -184,38 +244,51 @@
...
       'extension_requests:pp_cluster=B',
-    ],
-  )
+      ],
+    )
+  } else {
+    run_task('pe_xl::agent_install', $compiler_hosts,
+      server        => $master_host,
+      install_flags => [
+      '--puppet-service-ensure', 'stopped',
```
Comment (RandellP) [inline plans/install.pp]: indent array values.

**Review by RandellP — CHANGES_REQUESTED** [review]

File: plans/install.pp (hunk 4 above)
Comment (billclaytor) [reply plans/install.pp]: Updated

File: plans/install.pp (hunk 3 above)
Comment (billclaytor) [reply plans/install.pp]: Updated

File: plans/install.pp (hunk 5 above)
Comment (billclaytor) [reply plans/install.pp]: Updated

File: plans/install.pp (hunk 6 above)
Comment (billclaytor) [reply plans/install.pp]: Updated

**Review by RandellP — APPROVED** [review]

**Comment by billclaytor** [comment]
Closing this PR; will open a PR to `reidmv/reidmv-pe_xl`.

**Pull request closed without merging**